src/psadt-silent.template/templates/tools/chocolateyinstall.ps1

```diff
@@ -0,0 +1,67 @@
+﻿$ErrorActionPreference = 'Stop' # stop on all errors
+$toolsDir   = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+$zipArchive = Join-Path $toolsDir -ChildPath '[[Zip]]'
+$destination = (Join-Path $toolsDir -ChildPath $env:ChocolateyPackageName)
+
+Get-ChocolateyUnzip -FileFullPath $zipArchive -Destination $destination
+
+$adtFolder = 'AppDeployToolkit'
+$modulePath = Join-Path $destination -ChildPath "$adtFolder"
+$module = Join-Path $modulePath -ChildPath 'AppDeployToolkitMain.ps1'
+. $module
+$adtExe = Join-Path $destination -ChildPath "Deploy-Application.exe"
+$deployScript = Join-Path $destination -ChildPath 'Deploy-Application.ps1'
+
+try {
+    $packageArgs = @{
+        packagename = $env:ChocolateyPackageName
+        filetype = 'exe'
+        file = $adtExe
+        silentArgs = "-DeploymentType 'Install' -DeployMode 'Silent'"
+        validExitCodes = @(0)
+        softwareName = '[[Software]]*'
+    }
+
+    Install-ChocolateyInstallPackage @packageArgs
+} catch {
+    $packageArgs = @{

```

Comment:
What is the reason this catch block is needed? It seems to be doing pretty much the same thing as the try block.